server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -349,6 +363,8 @@ public boolean syncUpTo(long offset) throws IOException {
                         try {
                             outputStream.flush();
                             checkpointToSync = getCheckpoint();
+                            flushedSequenceNumbers = nonFsyncedSequenceNumbers;
+                            nonFsyncedSequenceNumbers = new LongArrayList();
```

Comment:
Should we carry over the current size of `nonFsyncedSequenceNumbers` as the estimate for the new list so we can reduce resizing?

Reply:
Please ignore this. I don't think it's a good suggestion.